style/scripts/taginfo.js

```diff
@@ -0,0 +1,59 @@
+"use strict";
+
+import * as fs from "fs";
+import * as ShieldDef from "../js/shield_defs.js";
+
+/**
+ * Adds documentation about network=* tags to a project description object, modifying it in place.
+ *
+ * @param {*} project - The project description object to modify.
+ * @param {*} sprites - Sprite metadata object parsed from a JSON spritesheet.
+ */
+function addNetworkTags(project, sprites) {
+  // Inject a map of each sprite ID to an absolute image URL instead of the usual sprite metadata.
+  let shieldImageURLs = Object.fromEntries(
+    Object.keys(sprites).map((sprite) => [
+      sprite,
+      `https://raw.githubusercontent.com/ZeLonewolf/openstreetmap-americana/main/style/icons/${sprite}.svg`,
+    ])
+  );
+  let shields = ShieldDef.loadShields(shieldImageURLs);
+
+  // Convert each shield's rendering metadata to an entry that taginfo understands.
+  let tags = Object.entries(shields).map((entry) => {
+    let network = entry[0],
+      definition = entry[1];
+
+    let description = `Roads carrying routes in this network are marked by shields`;
+    if (definition.modifiers && definition.modifiers.length > 0) {
+      description += ` modified by ${definition.modifiers.join(", ")} banners`;
+    }
+    description += ".";
+
+    let icon = definition.backgroundImage;
```

Comment:
For programmatically generated background images, such as calls to `circleShield()` and `roundedRectShield()`, the `icon_url` property ends up getting omitted from the JSON file because a function can’t be serialized to JSON. Ideally, we would save off the icon in a separate image in addition to the spritesheet so that we can link to it. But I think that can be tail work, because the icon is mostly decorative. Mappers don’t really need to know what the shield will look like, just that it’ll be present.